hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/util/ReflectionUtilsTest.java

```diff
@@ -50,9 +63,49 @@ public void readStaticFieldOrNull_readFromPublicField() {
         assertEquals("staticPublicFieldContent", field);
     }
 
+    @Test
+    public void shouldDiscoverAllNestedClasses() throws ClassNotFoundException {
```

Comment:
The `when_<PRECONDITION>_then_<POSTCONDITION>` name pattern is more useful. Here you could add `when_nestedClassesOf_then_`.